src/System.Text.Encoding.CodePages/src/System/Text/DecoderNLS.cs

```diff
@@ -45,6 +46,49 @@ internal DecoderNLS()
             Reset();
         }
 
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.AddValue(DecoderNLSSurrogate.EncodingKey, m_encoding);
+            info.AddValue(DecoderNLSSurrogate.DecoderFallbackKey, m_fallback);
+            info.SetType(typeof(DecoderNLSSurrogate));
+        }
+
+        [Serializable]
+        internal sealed class DecoderNLSSurrogate : IObjectReference, ISerializable
+        {
+            internal const string EncodingKey = "Encoding";
+            internal const string DecoderFallbackKey = "DecoderFallback";
+
+            private readonly Encoding _encoding;
+            private readonly DecoderFallback _fallback;
+
+            internal DecoderNLSSurrogate(SerializationInfo info, StreamingContext context)
+            {
+                if (info == null)
+                {
+                    throw new ArgumentNullException(nameof(info));
+                }
+                _encoding = (Encoding)info.GetValue(EncodingKey, typeof(Encoding));
+                _fallback = (DecoderFallback)info.GetValue(DecoderFallbackKey, typeof(DecoderFallback));
+            }
+
+            public object GetRealObject(StreamingContext context)
+            {
+                Decoder decoder = _encoding.GetDecoder();
+                if (_fallback != null)
+                {
+                    decoder.Fallback = _fallback;
+                }
+                return decoder;
+            }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext context)
+            {
+                // This should never be called
+                throw new NotImplementedException();
```

Comment:
No strong preference, as it should never be called :) I can change it to be a NotSupportedException; ok?